apps/src/componentLibrary/segmentedButtons/_SegmentedButton.tsx

```diff
@@ -0,0 +1,88 @@
+import React, {useCallback} from 'react';
+import classnames from 'classnames';
+import moduleStyles from '@cdo/apps/componentLibrary/segmentedButtons/segmentedButtons.module.scss';
+import FontAwesomeV6Icon from '@cdo/apps/componentLibrary/fontAwesomeV6Icon';
+
+type SegmentedButtonIconProps = {
+  iconName: string;
+  iconStyle: 'light' | 'solid' | 'regular' | 'thin';
+  title: string;
+};
+
+export type SegmentButtonType = 'withLabel' | 'iconOnly' | 'number';
```

Comment:
Yeah, that would be perfect. Still looking for a way to implement it though

Reply:
Just curious if we could subtype `_SegmentedButton` to `_SegmentedIconOnlyButton` and restrict there?

Reply:
Thanks for the hint, Alice!
Spent some time with ChatGpt and --worked out-- a solution for this.

Reply:
ok, it turned out that
1. That solution caused a lot of type errors in `SegmentedButtons.tsx` and `SegmentedButtons.story.tsx` .
I've tried to fix that, but didn't succed. I'll try again tomorrow, but not sure if that's possible. In case I won't found a quick solution for this - I suggest we revert last two commits in this PR (@levadadenys
[rework _segmentedButton types to be more strict and clear](https://github.com/code-dot-org/code-dot-org/pull/52357/commits/d0e313971eb995bc6c80339f99c16b32d53a13ce)
[d0e3139](https://github.com/code-dot-org/code-dot-org/pull/52357/commits/d0e313971eb995bc6c80339f99c16b32d53a13ce)
@levadadenys
[a try to fix type errors](https://github.com/code-dot-org/code-dot-org/pull/52357/commits/2b1174545e58bf9755c8c99074f6189e2238308f)
[2b11745](https://github.com/code-dot-org/code-dot-org/pull/52357/commits/2b1174545e58bf9755c8c99074f6189e2238308f) ) and merge this PR as it was before when this thread has started.

2. I think we should TypeScript type check as one of our pre-commit hooks, since I was able to commit files with type errors two times, one of them without knowing that I've broken anything.

Reply:
Thanks Denys! I'm definitely new to TS so not sure how to help with fixing these type errors. So reverting the last 2 commits SGTM!

Another question I just thought of is that are we assuming that a `SegmentedButtons` component would consist of a list of `SegmentedButton`s of the same `buttonType`. Or would it be okay to have a `SegmentedButtons` component which has mixed `buttonType`s - such as an icon with a number button?

Reply:
Yeah doing conditional type verification can get tricky. Might be better to just have some runtime warnings or even just clearer documentation what props are required for the different button types.

Reply:
+1 on adding clearer documentation - thanks!

Reply:
> Another question I just thought of is: are we assuming that a `SegmentedButtons` component would consist of a list of `_SegmentedButton`'s of the same `buttonType`? Or would it be okay to have a `SegmentedButtons` component which has mixed `buttonType`'s - such as an icon with a number button?

Yeah, all the buttons in `SegmentedButtons` should be of the same type, and the only way to set that type - is passing `type` prop to `SegmentedButtons` component. 

Commits that had the 'a try to make conditional types'  might have been a little bit confusing, please check again once I'll push the latest version of this pr